server/server.go

```diff
@@ -331,7 +333,8 @@ func newServer(cfg Config, metrics *Metrics) (*Server, error) {
 		}
 	}
 
-	logger.WithField("http", httpListener.Addr()).WithField("grpc", grpcListener.Addr()).Infof("server listening on addresses")
+	loggerWithFields := gokit_log.With(logger, "http", httpListener.Addr(), "grpc", grpcListener.Addr())
+	level.Info(loggerWithFields).Log("msg", "server listening on addresses")
```

Comment:
Looks a very convoluted way to do it. Do this instead:

```suggestion
	level.Info(loggerWithFields).Log("msg", "server listening on addresses", "http", httpListener.Addr(), "grpc", grpcListener.Addr())
```